### PR TITLE
Feature/sharing update to fluxc

### PR DESCRIPTION
**Pull request by nbradbury**
Updates the long-awaited sharing feature to support FluxC.

### CODE REVIEW

**Comment by nbradbury** [comment]
> why are you passing in a SiteModel to PublicizeWebView/Detail/ListFragment instead of the siteId?

I did that because it's the convention used elsewhere in the app, but you're right - we really only need to pass the site ID. Changed in 0ff514e.

File: WordPress/src/main/res/layout/publicize_list_fragment.xml [hunk 1]
```diff
@@ -51,7 +51,7 @@
                 android:textColor="@color/grey_dark"
                 android:textSize="@dimen/text_sz_medium" />
 
-            <org.wordpress.android.widgets.WPButton
+            <Button
```
Comment (nbradbury) [inline WordPress/src/main/res/layout/publicize_list_fragment.xml]: I'm not sure how that would affect the button's size, but that was a change you made here: 4590c8f08b7c37eaa942041ede9b3d9fce4dde36

**Comment by kwonye** [comment]
> I did that because it's the convention used elsewhere in the app

Oh, if that's the case I'm totally down with keeping it! I'm still unfamiliar with FluxC and would rather stay with convention than optimize a tiny model passing through.

File: WordPress/src/main/res/layout/publicize_list_fragment.xml (hunk 1 above)
Comment (kwonye) [inline WordPress/src/main/res/layout/publicize_list_fragment.xml]: Sorry, didn't mean size, I meant background. Yea, I made the change, just wanted to make sure it was conscious that it wasn't touched 👍

**Comment by kwonye** [comment]
:shipit:

**Comment by oguzkocer** [comment]
> why are you passing in a SiteModel to PublicizeWebView/Detail/ListFragment instead of the siteId?

I know this is already merged, but the idea behind passing the model instead of the `id` is to avoid confusion about if that's a local id or a remote one. So, if we do want to pass the id, we should name the variables to reflect their type. So, in this case using `ARG_SITE_REMOTE_ID` instead of `ARG_SITE_ID` would probably better. I hope this helps!

For reference, here are @maxme's comments on the subject: https://github.com/wordpress-mobile/WordPress-Android/pull/5856#discussion_r115507051 & https://github.com/wordpress-mobile/WordPress-Android/pull/5856#discussion_r115523625

**Comment by nbradbury** [comment]
> I know this is already merged, but the idea behind passing the model instead of the id is to avoid confusion about if that's a local id or a remote one. So, if we do want to pass the id, we should name the variables to reflect their type.

Makes perfect sense - I'll submit another PR which makes this change.

**Comment by maxme** [comment]
> ut the idea behind passing the model instead of the id is to avoid confusion about if that's a local id or a remote one

Yes exactly this. It's not a big deal but we should only see `id`s when we call the REST api and keep the model everywhere else.

**Comment by kwonye** [comment]
@oguzkocer @maxme thanks for the info! @nbradbury can you revert 0ff514e in your next PR please? Sorry about that.

**Pull request merged**